### PR TITLE
Feature/streaming hitl

### LINKED ISSUE #386: HITL for streaming
* Set up stream
* Parse a few incoming frames and verify that format and batch ID makes sense
* Disable stream

### PR DESCRIPTION
**Pull request by ryan-summers**
This PR fixes #386 by implementing HITL tests for the data livestream functionalities of Stabilizer.

- [x] This should be merged after https://github.com/quartiq/stabilizer/pull/425

### CODE REVIEW

**Comment by ryan-summers** [comment]
bors try

**Comment by ryan-summers** [comment]
bors try

**Comment by ryan-summers** [comment]
@jordens Do you have any ideas in regards to the shared code between this and `scripts/stream_throughput.py`? I'm not a fan of duplicating the code in both places, but I'm not sure what a good way to structure it would be otherwise.

File: hitl/streaming.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+def sequence_delta(previous_sequence, next_sequence):
+    """ Check the number of items between two sequence numbers. """
+    if previous_sequence is None:
+        return 0
+
+    delta = next_sequence - (previous_sequence + 1)
+    return delta & 0xFFFFFFFF
+
+
+class StabilizerStream:
```
Comment (ryan-summers) [inline hitl/streaming.py:45]: Should this be moved into a common `stabilizer` python package that we place somewhere else? Then, do some `from stabilizer import StabilizerStream`?

Comment (jordens) [reply hitl/streaming.py:45]: I'd leave it for now. Pretty sure we'll refactor/port/rewrite it when writing a web app or porting the stream decoding lib to rust.

Comment (jordens) [reply hitl/streaming.py:45]: But in light of the upcoming CLI tool for the iir filters, yes, make this `/py/stream.py` and add `/py/__init__.py`. The CLI tool would also go there.

Right: `streaming_throughput` and this hitls test should both use a common lib for stream decoding.

File: hitl/streaming.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    }
+
+    def __init__(self, port):
+        """ Initialize the stream. """
+        self.socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        self.socket.bind(("", port))
+        self.socket.settimeout(0.5)
+
+
+    def clear(self, duration=5):
```
Comment (jordens) [inline hitl/streaming.py:69]: Hmm. Just sleep 5 seconds and then do `while len(self.socket.recv(4096)): pass` to drain?

**Review by jordens — CHANGES_REQUESTED** [review]

File: hitl/run.sh [hunk 3]
```diff
@@ -11,10 +11,11 @@
 set -eux
 
 # Set up python for testing
-python3 -m venv --system-site-packages py
+python3 -m venv --system-site-packages py-venv
 . py/bin/activate
```
Comment (jordens) [inline hitl/run.sh:15]:
```suggestion
. py-venv/bin/activate
```

**Review by jordens — CHANGES_REQUESTED** [review]

**Comment by ryan-summers** [comment]
bors try

**Comment by ryan-summers** [comment]
bors try

**Review by jordens — APPROVED** [review]

**Comment by ryan-summers** [comment]
bors merge

**Pull request merged**